driver-core/src/main/java/com/datastax/driver/core/ExecutionInfo.java

```diff
@@ -112,8 +113,24 @@ public Host getQueriedHost() {
     }
 
     /**
+     * The speculative execution that completed this query.
+     * <p>
+     * 1 represents the initial, regular execution of the query, 2 represents the first speculative
+     * execution, etc.
+     * <p>
+     * Note that this is different from the number of <em>started</em> executions. For example, if
+     * one speculative execution was triggered, but the initial execution eventually completed
+     * first, this will be 1.
```

Comment:
I agree it would be nice to add the number some context onto at least which hosts were tried. I also understand why you avoided making this change as it's not trivial.  Mostly I think it's confusing to the user.

I think we need to be more explicit about the information we are returning on the ExecutionInfo. Without knowing how we handle speculative execution, I can see users being confused as to why speculative executions executed on hosts aren't listed in the triedhosts list for instance. Perhaps we could provide seperate specific speculative execution related information with regards to things like tried hosts.